v4/platformUI/src/main/java/exchange/dydx/platformui/components/textgroups/PlatformAutoSizingText.kt

```diff
@@ -0,0 +1,43 @@
+package exchange.dydx.platformui.components.textgroups
+
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.drawWithContent
+import androidx.compose.ui.text.TextStyle
+
+// Taken from https://stackoverflow.com/a/68258975
+@Composable
+fun PlatformAutoSizingText(
+    text: String,
+    textStyle: TextStyle,
+    modifier: Modifier = Modifier
+) {
+    var scaledTextStyle by remember { mutableStateOf(textStyle) }
+    var readyToDraw by remember { mutableStateOf(false) }
+
+
+    Text(
+        text,
+        modifier.drawWithContent {
+            if (readyToDraw) {
+                drawContent()
+            }
+        },
+        style = scaledTextStyle,
+        softWrap = false,
+        maxLines = 1,
+        onTextLayout = { textLayoutResult ->
+            if (textLayoutResult.hasVisualOverflow) {
+                scaledTextStyle =
+                    scaledTextStyle.copy(fontSize = scaledTextStyle.fontSize * 0.9)
```

Comment:
will this effectively recursively reduce by 10% until it fits?

Reply:
yup exactly, weirdly/unfortunately there doesn't seem to be an existing component/property that computes exact size to fit